src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/PostgresDirectoryDAO.scala

```diff
@@ -834,7 +834,13 @@ class PostgresDirectoryDAO(protected val dbRef: DbReference,
   override def setGoogleSubjectId(userId: WorkbenchUserId, googleSubjectId: GoogleSubjectId, samRequestContext: SamRequestContext): IO[Unit] = {
     runInTransaction("setGoogleSubjectId", samRequestContext)({ implicit session =>
       val u = UserTable.column
-      samsql"update ${UserTable.table} set ${u.googleSubjectId} = ${googleSubjectId} where ${u.id} = ${userId}".update().apply()
+      val updateGoogleSubjectIdQuery =
+        samsql"""update ${UserTable.table} set ${u.googleSubjectId} = ${googleSubjectId}
+                where ${u.id} = ${userId} and ${u.googleSubjectId} is null"""
+
+      if (updateGoogleSubjectIdQuery.update().apply() != 1) {
+        throw new WorkbenchException(s"Cannot overwrite an existing googleSubjectId for user ${userId}")
+      }
```

Comment:
Does `updateGoogleSubjectIdQuery.update().apply() ` return a count? If so, is there a realistic chance that the query count is 0 because the id doesn't exist for some reason?

Reply:
Yeah, `updateGoogleSubjectIdQuery.update().apply()` returns the count of rows that were updated by the query. So yes, it is entirely possible that the query count is 0 because the id doesn't exist for some reason. The service layer prevents this in practice, but a more accurate exception message here would be "Cannot update googleSubjectId for user ${userId} because user does not exist or the googleSubjectId has already been set". I'll update accordingly

Reply:
And while we're considering other values for that count, I'm not going to worry too much about `updateGoogleSubjectIdQuery.update().apply()` returning more than 1 since Postgres will enforce <=1 record for a given user Id